samples/features/sql-management-objects/src/CollectionSamples.cs

```diff
@@ -0,0 +1,64 @@
+﻿
+using System.Diagnostics;
+using Microsoft.SqlServer.Management.Smo;
+
+namespace Microsoft.SqlServer.SmoSamples
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using NUnit.Framework;
+    using Assert = NUnit.Framework.Assert;
+
+    [TestClass]
+    public class CollectionSamples
+    {
+        public VisualStudio.TestTools.UnitTesting.TestContext TestContext { get; set; }
+
+        [TestMethod]
+        public void Collection_iteration_is_faster_with_SetDefaultInitFields()

```

Comment:
So overall I like the idea of having actual examples users can run to see the benefits. But it'd definitely be great to also explain how this is actionable for them - ideally every test would have an explanation of what it's demonstrating and how users could use it in their own code.